great_expectations/rule_based_profiler/types/data_assistant_result/data_assistant_result.py

```diff
@@ -130,34 +135,44 @@ def get_line_chart(
         Returns:
             An altair line chart
         """
-        metric_title: str = metric_name.replace("_", " ").title()
-        domain_title: str = domain_name.title()
-
-        title: Union[str, alt.TitleParams] = f"{metric_title} per {domain_title}"
-        if subtitle:
-            title = alt.TitleParams(title, subtitle=[subtitle])
+        metric_plot_util: MetricPlotUtil = MetricPlotUtil(metric_name, metric_type)
+        domain_plot_util: DomainPlotUtil = DomainPlotUtil(domain_name, domain_type)
+        batch_id_plot_util: BatchIdPlotUtil = BatchIdPlotUtil(
+            "batch_id", AltairDataTypes.NOMINAL.value
+        )
+        return DataAssistantResult._get_line_chart(
+            df=df,
+            metric_plot_util=metric_plot_util,
+            domain_plot_util=domain_plot_util,
+            batch_id_plot_util=batch_id_plot_util,
+            subtitle=subtitle,
+        )
```

Comment:
Pattern I'm proposing works as follows - public methods take in the raw arguments, formats them into these utility classes, and passes them downstream to a private method to perform the actual logic. This keeps our API consistent and removes the need for the user to know about these dataclasses.